python/lsst/obs/base/mapper_tests.py

```diff
@@ -0,0 +1,207 @@
+        query_format: list
+            format list for the results portion of queryMetadata
+        queryMetadata: tuple of (dict, tuple)
+            dataIds and the results of calling them in queryMetadata
+        metadata_output_path: str
+            path to metadata output associated with dataIds['raw']
+        map_python_type: str
+            full python type specification returned by the mapper for dataIds['raw']
+        map_cpp_type: str
+            C++ type specification returned by the mapper for dataIds['raw']
+        map_storage_name: str
+            butler name for the storage type dataIds['raw']
+        raw_filename: str
+            Name of the raw files returned by the mapper for dataIds['raw']
+        default_level: str
+            value returned from mapper.getDefaultLevel
+        raw_levels: tuple of (str, set of str)
+            (level, expect) level and expected mapper return for mapper.getKeys('raw', level)
+        """
+        fields = ['output',
+                  'path_to_raw',
+                  'keys',
+                  'query_format',
+                  'queryMetadata',
+                  'metadata_output_path',
+                  'map_python_type',
+                  'map_cpp_type',
+                  'map_storage_name',
+                  'raw_filename',
+                  'default_level',
+                  'raw_levels',
+                  ]
+        MapperData = collections.namedtuple("MapperData", fields)
+        self.mapperData = MapperData(output=output,
+                                     path_to_raw=path_to_raw,
+                                     keys=keys,
+                                     query_format=query_format,
+                                     queryMetadata=queryMetadata,
+                                     metadata_output_path=metadata_output_path,
+                                     map_python_type=map_python_type,
+                                     map_cpp_type=map_cpp_type,
+                                     map_storage_name=map_storage_name,
+                                     raw_filename=raw_filename,
+                                     default_level=default_level,
+                                     raw_levels=raw_levels,
+                                     )
+
+    def tearDown(self):
+        del self.mapper
+        super(MapperTests, self).tearDown()
+
+    def test_map_config_data(self):
+        dataId = self.dataIds['raw']
+        butlerLocation = self.mapper.map_processCcd_config(dataId)
+        self.assertEqual(butlerLocation.getPythonType(), "lsst.pipe.tasks.processCcd.ProcessCcdConfig")
+        self.assertEqual(butlerLocation.getCppType(), "Config")
+        self.assertEqual(butlerLocation.getStorageName(), "ConfigStorage")
+        processCcd_path = os.path.join(self.mapperData.output, "config", "processCcd.py")
+        self.assertEqual(butlerLocation.getLocations(), [processCcd_path])
```

Comment:
No, it's not necessary. The top-level unittest asserts (e.g. assertEqual) are smart enough to output useful messages for sequences. Try this one to see:

```python
import unittest

class TestThing(unittest.TestCase):
    def test_thing1(self):
        self.assertEqual([1,2,3], [1,2,'b'])
    def test_thing2(self):
        self.assertListEqual([1,2,3], [1,2,'b'])

if __name__ == '__main__':
    unittest.main()
```